From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TheUnrealZaka <info@theunrealzaka.me>
Date: Tue, 10 Sep 2024 15:26:34 +0000
Subject: [PATCH] Disable player stats


diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index 9d7837f90340a96edbd544826ae2241c61ae72c4..4173a44e3f985152f554bdd55118da7370eaa900 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -2252,6 +2252,7 @@ public class ServerPlayer extends Player {
 
     @Override
     public void awardStat(Stat<?> stat, int amount) {
+        return; // Disable player stats
-        this.stats.increment(this, stat, amount);
-        this.level().getCraftServer().getScoreboardManager().forAllObjectives(stat, this, (scoreaccess) -> {
-            scoreaccess.add(amount);
-        });
     }
 
     @Override
